server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java

```diff
@@ -2824,4 +2824,23 @@ public void test_insert_on_conflict_dynamic_column_addition() {
         assertThat(response.cols()).containsExactly("a", "b", "c", "d", "e");
         assertThat(response).hasRows("1| 2| 3| NULL| 5", "2| NULL| NULL| 4| NULL");
     }
+
+    @Test
+    public void test_updating_generated_column_referencing_primary_key_column() {
```

Comment:
This test never failed on master, although `IndexOutOfBoundsException` was thrown... Any suggestions?

Reply:
can't we have a UnitTest for this case in the `IndexerTest`?
otherwise an option would be to check the `TransportReplicationAction.logger` to check there are no WARN messages (similar to what we test for the deprecation logger), but I think that's complex.

@seut Any other idea?

Reply:
> Do you we mean you saw an exception during the execution but statement didn't fail?

Yes, all statements and the test passed.

As Marios suggested, I added a unit test instead. Initially thought since the error came from the replica so an integration test was a must.